docs/scripting-element-object.rst

```diff
@@ -0,0 +1,566 @@
+then error happened during the operation; ``reason`` provides an information about error type;
+otherwise ``ok`` is ``true`` and the returned value is stored in the second variable.
+
+
+.. _splash-element-exists:
+
+element:exists
+--------------
+
+Check whether the element exists in DOM. If the element doesn't exist some of the methods will fail raising
+the error flag.
+
+**Signature:** ``exists = element:exists()``
+
+**Returns:** ``exists`` indicated whether the element exists.
+
+**Async:** no.
+
+
+.. _splash-element-mouse-click:
+
+
+element:mouse_click
+-------------------
+
+Trigger mouse click event on the element.
+
+**Signature:** ``ok, reason = element:mouse_click{x=0, y=0}``
+
+**Parameters:**
+
+* x - optional, x coordinate relative to the left corner of the element
+* y - optional, y coordinate relative to the top corner of the element
+
+**Returns:** ``ok, reason`` pair. If ``ok`` is nil then error happened during the
+function call; ``reason`` provides an information about error type.
+
+**Async:** no.
+
+If x or y coordinate is not provided they will be set to 0 and the click will be triggered
+on the left-top corner of the element. The coordinates can have a negative value which means
+the click will be triggered outside of the element.
+
+Mouse events are not propagated immediately, to see consequences of click
+reflected in page source you must call :ref:`splash-wait`
+
+See more about mouse events in :ref:`splash-mouse-click`.
+
+
+.. _splash-element-mouse-hover:
+
+element:mouse_hover
+-------------------
+
+Trigger mouse hover (JavaScript mouseover) event on the element.
+
+**Signature:** ``ok, reason = element:mouse_hover{x=0, y=0}``
+
+**Parameters:**
+
+* x - optional, x coordinate relative to the left corner of the element
+* y - optional, y coordinate relative to the top corner of the element
+
+**Returns:** ``ok, reason`` pair. If ``ok`` is nil then error happened during the
+function call; ``reason`` provides an information about error type.
+
+**Async:** no.
+
+If x or y coordinate is not provided they will be set to 0 and the click will be triggered
+on the left-top corner of the element. The coordinates can have a negative value which means
+the click will be triggered outside of the element.
+
+Mouse events are not propagated immediately, to see consequences of click
+reflected in page source you must call :ref:`splash-wait`
+
+See more about mouse events in :ref:`splash-mouse-click`.
+
+
+.. _splash-element-styles:
+
+element:styles
+--------------
+
+Return the computed styles of the element.
+
+**Signature:** ``styles = element:styles()``
+
+**Returns:** ``styles`` is a table with computed styles of the element.
+
+**Async:** no.
+
+Example of getting the font size of the element using this method.
+
+.. code-block:: lua
+
+    function main(splash)
+        local element = splash:select('.element')
+        return element:styles()['font-size']
+    end
```

Comment:
What's the difference between `element:styles()` and `element.node.style`? Could you please add it to docs?